issues/todo/ViMax_Complete_Integration_Package/src/pipelines/script2video_pipeline.py

```
@@ -0,0 +1,673 @@
+            #     with open(characters_path, "w", encoding="utf-8") as f:
+            #         json.dump([c.model_dump() for c in characters], f, ensure_ascii=False, indent=4)
+            #     print(f"☑️ Extracted {len(characters)} characters from script and saved to {characters_path}.")
+
+        if character_portraits_registry is None:
+            character_portraits_registry_path = os.path.join(self.working_dir, "character_portraits_registry.json")
+            if os.path.exists(character_portraits_registry_path):
+                with open(character_portraits_registry_path, "r", encoding="utf-8") as f:
+                    character_portraits_registry = json.load(f)
+                print(f"🚀 Loaded {len(character_portraits_registry)} character portraits from existing file.")
+            else:
+                print(f"🔍 Generating character portraits...")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)